### PR TITLE
Pl 71 abstract summary accordion item

### PR DESCRIPTION
**Pull request by matteovisotto**
It depends from PL 69 abstract accordion item

Done:
- Wait for abstract ready
- Use TextSummarizerService to summarize the abstract

### CODE REVIEW

File: src/components/SummaryKeyComponents/AbstractSummaryComponent.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    );
+  };
+
+  /**
+   * @private
+   * Callback for abstract ready event
+   * Summarize the abstract and load it into dom element
+   */
+  #summarizeAbstract = (abstract) => {
+    this.components.abstractSummaryText.innerText = textSummarizer(abstract, 6);
```
[the remark this answers is not in the document]
Comment (NinaGnjidic) [reply src/components/SummaryKeyComponents/AbstractSummaryComponent.js:42]: You really hate js, dont you 😆

[the remark this answers is not in the document]
Comment (tonesija) [reply src/components/SummaryKeyComponents/AbstractSummaryComponent.js:42]: hahahahhaahhaha, that comment was aimed at two problems

File: src/components/SummaryKeyComponents/AbstractComponent.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  EventHandlerService,
+  PDFLEvents,
+} from "../../services/EventHandlerService";
+
+/**
+ * Class representing the Abstract Accordion item
+ * @property {Object} components object that holds DOM elements that represent this component, as well as component's context
+ * @property {HTMLElement} components.abstractText html p element that represents the text of the accordion item
+ */
+class AbstractComponent {
```
Comment (NinaGnjidic) [inline src/components/SummaryKeyComponents/AbstractComponent.js:12]: Not really a meaningful name if you think about it, I know its long and ugly but something along the lines of AbstractAccordionItemComponent is maybe better?

File: src/components/SummaryKeyComponents/AbstractComponent.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    abstractText: document.querySelector("#abstractText"),
+  };
+
+  /**
+   * Set the pdf document needed to retrieve the Abstract
+   * Once the pdf is ready, start to fetch the Abstract
+   * @param pdfDoc the current pdf document
+   */
+  setPdf = (pdfDoc) => {
+    this.pdfDoc = pdfDoc;
```
Comment (NinaGnjidic) [inline src/components/SummaryKeyComponents/AbstractComponent.js:23]: Not documented.

**Review by NinaGnjidic — COMMENTED** [review]
As @tonesija stated in other similar PLs there's a part of code duplication in AbstractService which should be extracted. Also, as I suggested in those PLs maybe its better to put SummaryKeyComponent in SummaryKeyComponents folder.

**Pull request closed without merging**